testing/sdk_security.py

```diff
@@ -215,64 +219,79 @@ def setup_security(framework_name: str,
         role_name="test__integration__{}-role".format(framework_name),
-        service_account_name=service_account
+        service_account_name=service_account,
+        additional_permissions=additional_permissions
     )
     log.info("Finished setting up strict-mode security")
 
     return service_account_info
 
 
 def cleanup_security(framework_name: str,
+                     additional_permissions: List[dict]=[],
+                     linux_user: str="nobody",
                      service_account: str="service-acct",
                      service_account_secret: str="secret") -> None:
-
     if sdk_utils.is_strict_mode():
-        log.info("Cleaning up strict-mode security")
         revoke_permissions(
-            linux_user="nobody",
-            role_name="{}-role".format(framework_name),
-            service_account_name=service_account
+            linux_user=linux_user,
+            role_name='{}-role'.format(framework_name),
+            service_account_name=service_account,
+            additional_permissions=additional_permissions
+        )
+        revoke_permissions(
+            linux_user=linux_user,
+            role_name="slave_public%252F{}-role".format(framework_name),
+            service_account_name=service_account,
+            additional_permissions=additional_permissions
         )
         revoke_permissions(
-            linux_user="nobody",
-            role_name="test__integration__{}-role".format(framework_name),
-            service_account_name=service_account
+            linux_user=linux_user,
+            role_name='test__integration__{}-role'.format(framework_name),
+            service_account_name=service_account,
+            additional_permissions=additional_permissions
         )
 
     delete_service_account(service_account, service_account_secret)
 
-    log.info("Finished cleaning up strict-mode security")
+    log.info('Finished cleaning up strict-mode security')
 
 
-def security_session(framework_name: str) -> None:
+def security_session(framework_name: str,
+                     additional_permissions: List[dict]=[],
+                     linux_user: str="nobody",
+                     service_account: str="service-acct",
+                     service_account_secret: str="secret") -> None:
     """Create a service account and configure permissions for strict-mode tests.
 
     This should generally be used as a fixture in a framework's conftest.py:
 
     @pytest.fixture(scope='session')
     def configure_security(configure_universe):
-        yield from sdk_security.security_session(framework_name)
+        yield from sdk_security.security_session(framework_name, additional_permissions, 'nobody', 'service-acct')
     """
     try:
         is_strict = sdk_utils.is_strict_mode()
         if is_strict:
-            setup_security(framework_name)
+            setup_security(framework_name, additional_permissions, linux_user, service_account, service_account_secret)
```

Comment:
As mentioned above, we could return the list of permissions actually granted here and then explicitly remove them in the cleanup step below.